samtranslator/model/api/api_generator.py

```diff
@@ -513,16 +516,19 @@ def _construct_api_domain(self, rest_api, route53_record_set_groups):  # type: i
             basepath_mapping.Stage = ref(rest_api.logical_id + ".Stage")
             basepath_resource_list.extend([basepath_mapping])
         else:
-            for path in basepaths:
-                path = "".join(e for e in path if e.isalnum())
+            for basepath in basepaths:
+                # Remove possible leading and trailing '/' because a base path may only
+                # contain letters, numbers, and one of "$-_.+!*'()"
+                basepath = basepath.strip("/")
```

Comment:
The `strip_hyphens` ternary would be much clearer here, as this is dead code if it's `True`.